### ISSUE TITLE
BehaviorSpec: Received a failure event for test with unknown id 

### ISSUE THREAD

**Issue by barkakatib** [kotest/kotest#2755]
Kotest version: 5.0.3

There is an issue with Kotest BehaviorSpec. Duplicate `then` statement across multiple `when` in a Test File are no longer executed when running the BehaviorSpec UnitTests via the Android studio IDE. So when there are failing asserts in those duplicate 'then' clauses, one one of them is executed and the test results show green.

When the tests are executed in command line, it throws the following error  “Received a failure event for test with unknown id ”

I have created a small example code below to explain the issue further.

```
class ExampleTest : BehaviorSpec({

    given("inputs are set") {

        `when`("add 1 and 2") {
            val a = 1
            val b = 2

            val result = exampleFunc(a, b)

             then("result should add them") {
                 result shouldBe 3
             }
        }

        `when`("add 2 and 3") {
            val a = 2
            val b = 3

            val result = exampleFunc(a, b)

            then("result should add them") {
                result shouldBe null
                // this is a failing test, in IDE it is not executed and test results show green
            }
        }
    }
})

fun exampleFunc(a: Int, b: Int): Int = a + b
```

Output of IDE: there is only 1 then statement and the test result shows green.
<img width="642" alt="Screen Shot 2022-01-04 at 15 20 13" src="https://user-images.githubusercontent.com/90334824/148072832-6e55b54e-8c56-412f-b5e6-f85bc335914e.png">

Output from command line, failing test is detected but without proper cause

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:testDebugUnitTest'.
> Received a failure event for test with unknown id '1.5'. Registered test ids: '[1.1, 1.3, 1.2, :app:testDebugUnitTest, 1.6]'
```

**Comment by timrijckaert** [comment]
Might be related to Junit4 test runner, not reproducable with Junit5 runner

**Comment by timrijckaert** [comment]
Not stale

**Comment by timrijckaert** [comment]
Not stale

**Comment by sksamuel** [comment]
Does this still happen with 5.7.1 ?

**Comment by sksamuel** [comment]
I believe this has been fixed since 5.7.x so closing. Please reopen if it reoccurs.
